src/stream/stream/last.rs

```diff
@@ -0,0 +1,42 @@
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct LastFuture<S, T> {
+    stream: S,
+    last: Option<T>,
+}
+
+impl<S, T> LastFuture<S, T> {
+    pin_utils::unsafe_pinned!(stream: S);
+    pin_utils::unsafe_pinned!(last: Option<T>);
```

Comment:
If you make `last` unpinned, the code will compile:

```suggestion
    pin_utils::unsafe_unpinned!(last: Option<T>);
```

This is okay to do because `last` is never really pinned.

Reply:
Nice, thanks!